SEM.09: Add 01-02 | Prac.07 Add 05 | Maksim Hristov

MaximHr opened the pull request:
Maksim Hristov
SI 4 4MI0600466

MaximHr committed:
'week-4-task15-16'

MaximHr committed:
'week4-prac'

MaximHr committed:
'deleted'

MaximHr committed:
'prac4'

MaximHr committed:
Merge branch 'nikola354:main' into main

MaximHr committed:
'separate-files'

MaximHr committed:
Merge branch 'nikola354:main' into main

MaximHr committed:
Merge branch 'nikola354:main' into main

MaximHr committed:
'Sem09:Task1'

MaximHr committed:
Revert "'Sem09:Task1'"

This reverts commit f34670490cd8c639232b82020a03d40734de1e21.

MaximHr committed:
'Sem09:Task1'

MaximHr committed:
'Sem09:Task2'

MaximHr committed:
'Prac07:Task5'

MaximHr committed:
'bug-fix'

MaximHr committed:
Delete Sem. 04/Pract.04/task17.cpp

Slavi15 commented on Sem. 09/Task2.cpp:
```diff
@@ -0,0 +1,43 @@
+void toUpper(char *str) {
+  if(!str) return;
+  while(*str != '\0') {
+    if(*str >= 'a' && *str <= 'z') {
```
On lines 19 and 28, instead of checking for lowercase or uppercase this way, it is better to create functions `bool isLower(char ch)` and `bool isUpper(char ch)`

Slavi15 commented on Sem. 09/Task2.cpp:
```diff
@@ -0,0 +1,43 @@
+//Задача 2 Напишете функции toUpper и toLower, които приемат стринг и променят всички главни букви в малки/всички малки букви в главни.
```
In the provided problem we meant converting all uppercase letters to lowercase and vice-versa simultaneously i.e

**Example:**

```c++
"HeLlo WoRLd"
```

```c++
"hElLO wOrlD"
```

Slavi15 commented on Sem. 09/Task1.cpp:
```diff
@@ -0,0 +1,60 @@
+    if(arr[i] == searched[0]) {
+      int j = 0;
+      bool isMatch = true;
+      while(searched[j] != '\0') {
```
On `line 26` reuse the function `bool isPrefix(const char* pattern, const char* text)`. Do not make inner loops without encapsulating the logic into a new function! The same applies to the while-loop on `line 36`.

Slavi15 commented on Sem. 07/Pract.07/Task5.cpp:
```diff
@@ -0,0 +1,56 @@
+#include <iostream>
+constexpr int SIZE = 10;
+
+void spiralPrint(const int arr[][SIZE], int size) {
```
Kudos! Very creative solution!

MaximHr committed:
'task2-enhancement'

MaximHr committed:
'separating-logic'

MaximHr committed:
Merge branch 'main' of https://github.com/MaximHr/Introduction-to-programming-FMI

Slavi15 committed:
Update and rename Task1.cpp to Task01.cpp

The provided solution was working, but was somewhat confusing with all the different indices! I did a refactor on your code by using the fundamental string manipulation functions, please peruse over the code and ask if you have any questions! Also, when printing strings, **do not iterate** through the whole char array, but just do `std::cout << str << std::endl`, because in strings, we have terminating zero which tells us where the end of the string is (as a sentinel)!

Slavi15 committed:
Update and rename Task2.cpp to Task02.cpp

Do not make functions for printing strings, just `std::cout << str << std::endl`

Slavi15 committed:
Update and rename Task5.cpp to Task05.cpp
